src/guides/arguments.md

```diff
@@ -0,0 +1,31 @@
+## Argument Flags :
+Function argument flags mark how arguments within a function may be used (or hence, not used at all), it will be displayed near or bleow the function argument as example :
+(- `userID` `(Flag : Required)`) etc
+
+- __Optional__ - The argument **can** be excluded but **cannot** be left empty, they show up as `(something)` and will **always** be after all required arguments.
+- __Required__ - The argument **must** be included and **cannot** be empty.
+- __Emptiable__ - The argument **must** be included but **can** be empty.
+- __Vacantable__ - The argument **can** be excluded and if included **can** be empty.
+
+
+## Argument Types :
+This section will explain the various argument types used for function arguments. Function arguments are anything that goes in the brackets `[]` of a function, it will be displayed near or bleow the function argument as example :
+(- `userID` `(Type : Snowflake || Flag : Required)`) and if it has more than one option it will be `(Type : String, Snowflake || Flag : Required)`etc
```

Comment:
```suggestion
## Argument Types
This section will explain the various argument types used for function arguments. Function arguments are anything that goes in the brackets `[]` of a function, it will be displayed near or below the function argument.
Example: (- `userID` `(Type: Snowflake || Flag: Required)`) and if it has more than one option it will be `(Type: String, Snowflake || Flag: Required)` etc.
```